apps/remote_control/lib/lexical/remote_control/modules.ex

```diff
@@ -47,8 +44,135 @@ defmodule Lexical.RemoteControl.Modules do
     end
   end
 
+  @moduledoc """
+  Utilities for dealing with modules on the remote control node
+  """
+
+  alias Future.Code.Typespec
+
+  @typedoc "Module documentation record as defined by EEP-48"
+  @type docs_v1 :: tuple()
+
+  @typedoc "A type, spec, or callback definition"
+  @type definition ::
+          {name :: atom(), arity :: arity(), formatted :: String.t(), quoted :: Macro.t()}
+
   @cache_timeout Application.compile_env(:remote_control, :modules_cache_expiry, {10, :second})
 
+  @doc """
+  Ensure the given module is compiled, returning the BEAM object code if successful.
+  """
+  @spec ensure_beam(module()) ::
+          {:ok, beam :: binary()} | {:error, reason}
+        when reason:
+               :embedded
+               | :badfile
+               | :nofile
+               | :on_load_failure
+               | :unavailable
+               | :get_object_code_failed
+  def ensure_beam(module) when is_atom(module) do
+    with {:module, _} <- Code.ensure_compiled(module),
```

Comment:
The idea is that this allows doc fetching to operate independently of compilation. I believe it was originally needed because there were situations in tests where a newly introduced module was loaded but not compiled, but I've since refactored the hover tests and this may not be needed anymore.